src/main/webapp/script-auth.js

```diff
@@ -0,0 +1,72 @@
+// firebase configuration details (nonsecret)
+var firebaseConfig = {
+    apiKey: "AIzaSyDkYtj0ZlLE4ggunFI8c6ov_wnVuDUsY6Y",
+    authDomain: "wear-systems-interns-step-2020.firebaseapp.com",
+    databaseURL: "https://wear-systems-interns-step-2020.firebaseio.com",
+    projectId: "wear-systems-interns-step-2020",
+    storageBucket: "wear-systems-interns-step-2020.appspot.com",
+    messagingSenderId: "372274219809",
+    appId: "1:372274219809:web:5eba81e89b21d1355e4c60",
+    measurementId: "G-EHL6SKF61R"
+};
+
+// initialize firebase
+function authInitializeFirebase() {
+    firebase.initializeApp(firebaseConfig);
+    firebase.analytics();
+}
+
+// check if user is logged in and show buttons accordingly
+function authCheckLogin() {
+    firebase.auth().onAuthStateChanged(function(user) {
+        if (user) {
+            document.getElementById('loginBtn').style.display = 'none';
+            document.getElementById('logoutBtn').style.display = 'block';
+        } else {
+            document.getElementById('loginBtn').style.display = 'block';
+            document.getElementById('logoutBtn').style.display = 'none';
+        }
+    });
+}
+
+// sign the user out
+async function authLogout() {
+    try {
+        await firebase.auth().signOut();
+        authCheckLogin();
+    }
+    catch(error) {
+        console.log(error);
+    }
+}
+
+/** SLIGHTLY MODIFIED FUNCTIONS FOR TESTING */
```

Comment:
Could you explain more why this is needed?  Also looks like the actual code is duplicated so if it is needed could you create a helper function for that?

Reply:
@ShreeragJ In these functions, I'm passing in the mock functions to be used so that the tests don't call the actual functions whereas in the non-test methods, I just call the actual functions directly.

Reply:
Regarding duplicated code, my original thought was that I'd need it to be able to call different functions and optionally pass in mock functions, though perhaps that could be implemented as a flag indicating as to whether it's being called for testing.

Reply:
Can you just have a single function and you either pass in mockFn or firebase.auth() depending on if you're in a test or not?  Having test-specific functions should be a very last resort, seems like we can do better here.

Reply:
Gotcha, will make those changes